test/cpp/ext/otel/otel_plugin_test.cc

```diff
@@ -1438,40 +1423,49 @@ TEST_F(OpenTelemetryPluginNPCMetricsTest, SetInt64Gauge) {
   auto stats_plugins =
       grpc_core::GlobalStatsPluginRegistry::GetStatsPluginsForChannel(
           grpc_core::StatsPlugin::ChannelScope("dns:///localhost:8080", ""));
-  stats_plugins.SetGauge(handle, kInitialValue, kLabelValues,
-                         kOptionalLabelValues);
+  stats_plugins.SetGauge(handle, kInitialValue, kLabelValuesSet1,
+                         kOptionalLabelValuesSet1);
   auto data = ReadCurrentMetricsData(
       [&](const absl::flat_hash_map<
           std::string,
           std::vector<opentelemetry::sdk::metrics::PointDataAttributes>>&
               data) { return !data.contains(kMetricName); });
-  EXPECT_THAT(data, ::testing::ElementsAre(::testing::Pair(
-                        kMetricName, ::testing::ElementsAre(::testing::AllOf(
-                                         AttributesEq(kLabelKeys, kLabelValues,
-                                                      kOptionalLabelKeys,
-                                                      kOptionalLabelValues),
-                                         GaugeResultEq(kInitialValue))))));
+  EXPECT_THAT(
+      data, ::testing::ElementsAre(::testing::Pair(
+                kMetricName,
+                ::testing::ElementsAre(::testing::AllOf(
+                    AttributesEq(kLabelKeys, kLabelValuesSet1,
+                                 kOptionalLabelKeys, kOptionalLabelValuesSet1),
+                    GaugeResultIs(IntOrDoubleEq(kInitialValue)))))));
```

Comment:
We probably no longer need `IntOrDoubleEq()`, because we know whether it's an int or double when we invoke the matcher.

Reply:
Can we use this same pattern in the other cases, so that we don't need it anywhere?  Or are there callers that really don't know the type they are matching against?